regression-test/suites/correctness_p0/test_xor_pred.groovy

```diff
@@ -0,0 +1,42 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_xor_pred") {
+    sql """ drop table if exists dbxor; """
+    sql """
+    CREATE TABLE IF NOT EXISTS dbxor (
+              `k1` INT(11) NULL COMMENT "",
+              `k2` INT(11) NULL COMMENT ""
+            ) ENGINE=OLAP
+            DUPLICATE KEY(`k1`)
+            DISTRIBUTED BY HASH(`k1`) BUCKETS 1
+            PROPERTIES (
+            "replication_allocation" = "tag.location.default: 1",
+            "storage_format" = "V2"
+    );
+    """
+    sql """set enable_nereids_planner=true,enable_fold_constant_by_be = false; """
+    qt_sql_const """
+        select 1 xor 1,0 xor 1,1 xor 0,0 xor 0,null xor 0,null xor 1,1 xor null ,0 xor null;
```

Comment:
what is the result of  `select 2 xor 2` ?

Reply:
will cast Boolean

Reply:
> will cast Boolean

you mean `select xor(2, 1)` -> 10 & 01 -> 11 -> true?

Reply:
xor(2,1) -> xor(cast(2 as bool), cast(1 as bool)) -> xor(true,true)

Reply:
> > will cast Boolean
> 
> you mean `select xor(2, 1)` -> 10 & 01 -> 11 -> true?

FunctionSignature.ret(BooleanType.INSTANCE).args(BooleanType.INSTANCE, BooleanType.INSTANCE));

Reply:
the signeture of xor is `boolean xor(boolean, boolean)`, maybe select `select xor(2,1)` in this implementation will be processed like 
`select xor(cast(2 as boolean), case(1 as boolean))`, which will get false as result.

Reply:
> xor(2,1) -> xor(cast(2 as bool), cast(1 as bool)) -> xor(true,true)

so `select xor(2, 1)` will get false. OK, that correct https://dbfiddle.uk/6lWP9wVP